site/_posts/2023-07-11-get-more-value-out-of-fusionauth.md

```diff
@@ -0,0 +1,321 @@
+
+We can harness the full power of FusionAuth in our application, hand off the user maintenance to the power users, and make it seamless for them. The application will need to ensure the user enters the correct information (**Validation**), guide them through setting up the new user correctly (**Workflow**), and we want to make the process feel like it is part of our application (**Branding**). We can accomplish all these through the use of the FusionAuth APIs.
+
+The examples below are written with the assumption that you are familiar with how to make REST calls using the [HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-7.0) in C#, the use of [asynchronous](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/) calls and Windows Forms applications.  In order for the sample application to run you must have a [working FusionAuth server](/docs/v1/tech/5-minute-setup-guide). Even if you are not totally familiar with these concepts, the snippets should be easy enough to follow with general programming knowledge. If you need a refresher on some of these topics, there are some links at the end.
+
+<br>
+
+## Custom Validation with FusionAuth APIs
+*“With great power, comes great responsibility!”*
+
+One of FusionAuth’s strengths is its flexibility. However, with the flexibility comes some complexity. To create a user in the FusionAuth front-end, you only need one piece of information. You will need to supply either the username or the email. (If you are using multiple tenants, you will need the tenant ID as well.) The front end gives you other optional fields to fill in as well, however none are required. When using the [Create a User API](/docs/v1/tech/apis/users#create-a-user), only the username or email is still required but there are also over 30 additional request parameters available for customization. Creating a custom front end around the APIs will not only allow you to choose which of the username or email fields to use, but also what other meta-data is appropriate and required for your application. Below is sample code for creating a user using the FusionAuth API.
+
+<br>
+
+### **Setup**
+First, we need to create a few objects to hold some values. By constructing the classes and decorating properties with the proper attributes, we can still use the strongly typed objects and standard naming conventions in our C# application while also using the JSON Serializer to easily format the request body when it is time.
+
+```cs
+//For User object
```

Comment:
Let's pull the code in via the `remote_include` tag from your sample gh repo. see site/_posts/2023-04-26-nextjs-single-sign-on.md for an example.

Reply:
You also should link to the GH repo at the beginning and say something like "If you'd like to just jump into the code, here's the repo".

Reply:
Is there a way to only select part of the code in a file.  I found something on tags and the | (pipe) but didn't seem to work.  I think I can re-work the app to make it a little more clean if not.

Reply:
Since we cannot include snippets until we move to astro, I edited the application to break down the code in a bit more modular fashion and have included as remote_includes.  For others, I thought it looked too cluttered and made it hard to follow.  For those I commented the code for the includes to the snippets when we convert to astro.